### PR TITLE
feat(@desktop/wallet): Update activity feed sections date markers

### LINKED ISSUE #11306: Handle date markers
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=13138-409539&mode=design&t=gC1yn1CBi7BUCE0O-4

### PR DESCRIPTION
**Pull request by Cuteivist**
closes #11306 

### What does the PR do

* Uses new time markers for sections
* Fixed timestamp in activity delegate

### Affected areas

Activity UI

### Screenshot of functionality (including design for comparison)

Design: https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=13138-409539&mode=design&t=O5H6uMt7DdkObklq-4

https://github.com/status-im/status-desktop/assets/11396062/f870c1e5-9c2e-4a40-b3ee-9bb34ab0f8be

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml [hunk 1]
```diff
@@ -236,9 +236,16 @@ QtObject {
 
     // return full days between 2 dates
     function daysBetween(firstDate, secondDate) {
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml:238]: You're changing the semantics of an existing function... don't like that :)

Better keep this one intact and create a new one instead

File: ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml [hunk 2]
```diff
@@ -288,13 +295,20 @@ QtObject {
         return value.toLocaleString(loc, formatString)
     }
 
+    function getFirstDayOfTheWeek() {
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml:298]: Some docu would be nice, it's not clear from the function name what it does

File: ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml [hunk 3]
```diff
@@ -288,13 +295,20 @@ QtObject {
...
+        const loc = Qt.locale()
+        let currDate = new Date()
+        const dayOffset = loc.firstDayOfWeek === Qt.Monday ? 1 : 0
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml:301]: This is wrong, what if the first day of week is Tuesday?

Comment (caybro) [reply ui/StatusQ/src/StatusQ/Core/LocaleUtils.qml:301]: See `Qt.locale().firstDayOfWeek`

File: ui/imports/shared/views/HistoryView.qml [hunk 4]
```diff
@@ -101,11 +101,33 @@ ColumnLayout {
...
+                        } else if (daysDiff < 2) {
+                            return qsTr("Yesterday")
+                        } else if (daysToBeginingOfThisWeek >= 0) {
+                            return qsTr("Earlier this week")
+                        } else if (daysToBeginingOfThisWeek > -7) {
+                            return qsTr("Last week")
+                        } else if (currDate.getMonth() === timestampDate.getMonth()) {
+                            return qsTr("Earlier this month")
+                        } else if ((new Date(new Date().setDate(0))).getMonth() === timestampDate.getMonth()) {
+                            return qsTr("Last month")
```
Comment (caybro) [inline ui/imports/shared/views/HistoryView.qml:128]: We already have very similar `LocaleUtils.formatRelativeTimestamp()`

**Review by caybro — CHANGES_REQUESTED** [review]

**Review by caybro — APPROVED** [review]
LGTM

thanks for the fixes!

**Pull request merged**